sql-ballerina/Module.md

```diff
@@ -0,0 +1,302 @@
+   //Can perform any operations using 'student' and can access any fields in the returned record of type Student.
+});
+```
+
+Defining the return type is optional and you can query the database without providing the result type. Hence, 
+the above example can be modified as follows with an open record type as the return type. The property name in the open record 
+type will be the same as how the column is defined in the database. 
+
+```ballerina
+// Select the data from the database table. The query parameters are passed 
+// directly. Similar to the `execute` examples, parameters can be passed as 
+// sub types of `sql:TypedValue` as well.
+int id = 10;
+int age = 12;
+sql:ParameterizedQuery query = `SELECT * FROM students
+                                WHERE id < ${id} AND age > ${age}`;
+stream<record{}, sql:Error> resultStream = dbClient->query(query);
+
+// Iterating the returned table.
+error? e = resultStream.forEach(function(record{} student) {
+    //Can perform any operations using 'student' and can access any fields in the returned record.
+});
+```
+
+There are situations in which you may not want to iterate through the database and in that case, you may decide
+to only use the `next()` operation in the result `stream` and retrieve the first record. In such cases, the returned
+result stream will not be closed, and you have to explicitly invoke the `close` operation on the 
+`sql:Client` to release the connection resources and avoid a connection leak as shown below.
+
+```ballerina
+stream<record{}, sql:Error> resultStream = 
+            dbClient->query("SELECT count(*) as total FROM students");
+
+record {|record {} value;|}|error? result = resultStream.next();
+
+if result is record {|record {} value;|} {
+    //valid result is returned.
+} else if result is error {
+    // An error is returned as the result.
+} else {
+    // Student table must be empty.
+}
+
+error? e = resultStream.close();
+```
+
+#### Updating Data
+
+This example demonstrates modifying data by executing an UPDATE statement via the `execute` remote function of 
+the client.
+
+```ballerina
+int age = 23;
+sql:ParameterizedQuery query = `UPDATE students SET name = 'John' 
+                                WHERE age = ${age}`;
+sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+```
+
+#### Deleting Data
+
+This example demonstrates deleting data by executing a DELETE statement via the `execute` remote function of 
+the client.
+
+```ballerina
+string name = "John";
+sql:ParameterizedQuery query = `DELETE from students WHERE name = ${name}`;
+sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+```
+
+#### Batch Updating Data
+
+This example demonstrates how to insert multiple records with a single INSERT statement that is executed via the 
+`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and 
+parameterized SQL query as same as the  above `execute` operations.
```

Comment:
```suggestion
a parameterized SQL query as same as the above `execute` operations.
```

Reply:
Change Added in https://github.com/ballerina-platform/module-ballerina-sql/pull/228/commits/2656f12c62276e15a256c133465cb60304d0b041